include/novatel_edie/decoders/common/json_reader.hpp

```diff
@@ -258,52 +249,42 @@ struct BaseField
     void SetConversion(const std::string& sConversion_)
     {
         conversion = sConversion_;
-        ParseConversion(sConversionStripped, conversionBeforePoint, conversionAfterPoint);
-        conversionHash = CalculateBlockCrc32(sConversionStripped.c_str());
-    }
 
-    void ParseConversion(std::string& strStrippedConversionString_, int32_t& iBeforePoint_, int32_t& iAfterPoint_) const
-    {
         const char* sConvertString = conversion.c_str();
-        int32_t* iSelectedPoint = &iBeforePoint_;
 
-        while (*sConvertString != 0)
+        if (*sConvertString != '%') { throw std::runtime_error("Encountered an unexpected character in conversion string"); }
+
+        ++sConvertString;
+
+        if (std::isdigit(*sConvertString))
         {
-            // "0x" could be a prefix on a hex field (0x%...) or a valid conversion string (%0x).
-            // Prevent these two cases from occurring at the same time.
-            if ((0 == memcmp(sConvertString, "0x", 2)) && (0 != strcmp(sConvertString, "0x"))) { sConvertString += 2; }
-
-            // If the value "10" or greater is found from the conversion string, two bytes would
-            // need to be consumed from the string to move past that value. Otherwise, only one byte
-            // is necessary to consume.
-            if (*sConvertString >= '0' && *sConvertString <= '9')
-            {
-                if (sscanf(sConvertString, "%d.", iSelectedPoint) != 1) { throw std::runtime_error("Failed to parse integer value"); }
-                sConvertString = sConvertString + (*iSelectedPoint > 9 ? 2 : 1); // Skip the numerals
-            }
+            conversionBeforePoint = std::stoi(sConvertString);
```

Comment:
This is a pretty significant change. You're sure its functionally equivalent?

Reply:
It depends on what kind of conversion strings are valid. I think the way we were doing this before with the loop was nonsensical and bugprone. The way I wrote it, it assumes that a valid conversion string must come in the form of %(x)(.y)a with () denoting optional fields. The way we had it before, there were a lot of seemingly faulty conversion strings that would pass.